packages/nimble-components/src/toggle-button/styles.ts

```diff
@@ -10,23 +16,66 @@ export const styles = css`
     .control[aria-pressed='true'] {
         background-color: ${fillSelectedColor};
         border-color: ${fillSelectedColor};
+        position: relative;
+        transition: box-shadow ${smallDelay};
+        transition: border-color ${smallDelay};
     }
 
     .control[aria-pressed='true']:hover {
-        background-color: ${fillSelectedColor};
+        border-color: ${borderHoverColor};
+        box-shadow: 0px 0px 0px 1px ${borderHoverColor} inset;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control[aria-pressed='true']${focusVisible} {
+        border-color: ${borderHoverColor};
+        box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+    }
+
+    .control[aria-pressed='true']:active {
+        box-shadow: none;
+        outline: none;
+        transition: outline 1s;
+        transition: box-shadow ${mediumDelay};
+    }
+
+    .control::before[aria-pressed='true'] {
+        content: '';
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
+        box-sizing: border-box;
+        outline: none;
+        background-clip: content-box;
+        z-index: -1;
+    }
+
+    .control::before[aria-pressed='true']:hover {
         background-color: ${fillSelectedColor};
     }
 
-    :host([disabled]) .control[aria-pressed='true'] {
+    .control::before[aria-pressed='true']${focusVisible} {
         background-color: ${fillSelectedColor};
+        outline: ${borderWidth} solid ${borderHoverColor};
+        outline-offset: -3px;
+        transition: outline 0.5s;
+        color: transparent;
+        padding: 2px;
+    }
+
+    .control::before[aria-pressed='true']:active {
+        outline: none;
+        padding: 1px;
+    }
+
+    :host([disabled]) .control[aria-pressed='true'] {
         border-color: ${fillSelectedColor};
+        background-color: ${fillSelectedColor};
     }
 
     :host([disabled]) .control[aria-pressed='true']:hover {
-        background-color: ${fillSelectedColor};
         border-color: ${fillSelectedColor};
+        background-color: ${fillSelectedColor};
     }
```

Comment:
The selection target of this element is `.control` and should be grouped with the other `.control` targetted selectors.